src/main/java/com/github/cypher/model/Client.java

```diff
@@ -173,9 +177,28 @@ public User getUser(String id) {
 		return userRepository.get(id);
 	}
 
-	private void addServer(String server) {
-		//Todo
-		servers.add(new Server(server));
+	private void addServer(String serverAddress) {
+		for (Server server:servers){
+			if (server.getAddress().equals(serverAddress)){
+				return;
+			}
+		}
+
+		Server server = new Server(serverAddress);
+		servers.add(server);
+
+
+		for (Room room: genCollection.getRoomsProperty().toArray(new Room[genCollection.getRoomsProperty().size()])) {
+			distributeRoom(room);
+		}
+		for (Room room: pmCollection.getRoomsProperty().toArray(new Room[pmCollection.getRoomsProperty().size()])) {
+			distributeRoom(room);
+		}
+		for (Server s: servers){
```

Comment:
Why not iterate over roomCollections rather than separating genCollection & pmCollection?